library/core/src/main/java/com/google/android/exoplayer2/text/cea/Cea608Decoder.java

```diff
@@ -323,19 +325,24 @@ protected void decode(SubtitleInputBuffer inputBuffer) {
         continue;
       }
 
+      // If we've reached this point then there is data to process; flag that work has been done.
+      captionDataProcessed = true;
+
       boolean previousCaptionValid = captionValid;
       captionValid = (ccHeader & CC_VALID_FLAG) == CC_VALID_FLAG;
       if (!captionValid) {
         if (previousCaptionValid) {
           // The encoder has flipped the validity bit to indicate captions are being turned off.
           resetCueBuilders();
-          captionDataProcessed = true;
         }
         continue;
       }
 
-      // If we've reached this point then there is data to process; flag that work has been done.
-      captionDataProcessed = true;
+      if (dataProcessedSinceLastRepeatableControl) {
```

Comment:
Do you understand what this bit of existing code is doing (which is kept in this change; I'm posting this comment here because GitHub doesn't let me comment on the line I want to comment on):

https://github.com/google/ExoPlayer/blob/71f72c59537711399dc5496136dd6b867acc6d77/library/core/src/main/java/com/google/android/exoplayer2/text/cea/Cea608Decoder.java#L329

I can't figure out what it's supposed to be doing. Perhaps it was an attempt to implement what you're now doing properly, and so becomes unnecessary?

Either way, does it work to get rid of both that bit of code and `dataProcessedSinceLastRepeatableControl`, and just do this here:
```
boolean repeatedControlPossible = repeatableControlSet;
repeatableControlSet = false;
```
and then pass `repeatedControlPossible` as an argument to `handleCtrl`? The line that sets `repeatableControlSet = false` inside `handleCtrl` would also go away.

Reply:
* ```repeatableControlSet``` seems to only save the value between 2 iterations of the loop.
* This part is probably only disabling the check for duplications between two calls of ```decode()```:
```
    if (!isRepeatableControl) {
        repeatableControlSet = false;
      }
```
I do not see why that is necessary at the moment. If 608 bytes are the first ones in each packet (as 608 bytes have fixed bandwidth), then 708 bytes are next to each other, then 2 calls to ```decode()``` probably cannot contain originally consecutive bytes in the analogue CEA stream.


Your proposal seems logical, I do not seem any drawbacks of that solution, let me update the code.

(Unfortunately, I cannot run full test cycles including various live content, sarnoff test cases etc. for every code change separately, but I will run them as soon as we have more - maybe all changes merged.)